107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java

```diff
@@ -78,6 +78,17 @@ public CacheManager getCacheManager(URI uri, ClassLoader classLoader, Properties
     Eh107CacheManager cacheManager;
     ConcurrentMap<URI, Eh107CacheManager> byURI;
 
+    Configuration config;
+    try {
+      if (URI_DEFAULT.equals(uri)) {
+        config = new DefaultConfiguration(classLoader);
+      } else {
+        config = new XmlConfiguration(uri.toURL(), classLoader);
+      }
+    } catch (Exception e) {
+      throw new javax.cache.CacheException(e);
+    }
+
     synchronized (cacheManagers) {
```

Comment:
The combined proposed changes would allow this whole logic to move to the method `createCacheManager(URI, Configuration, Properties)`